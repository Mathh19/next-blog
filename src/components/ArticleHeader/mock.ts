import { ArticleHeaderProps } from '.';

export default {
  title:
    'Sint deserunt officia sit ipsum. Minim commodo sit ea labore sit sit ex ad tempor esse labore.',
  excerpt:
    'Proident sunt est enim fugiat sit incididunt sunt veniam adipisicing amet aliquip.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleHeaderProps;
