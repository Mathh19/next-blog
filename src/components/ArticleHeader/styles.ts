import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const CoverContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    margin-bottom: ${theme.spacings.medium};
  `}
`;

export const Cover = styled.img`
  ${css`
    max-width: 100%;
    display: block;
  `}
`;
