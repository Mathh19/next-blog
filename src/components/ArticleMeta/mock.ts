import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '1234567890',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleMetaProps;
