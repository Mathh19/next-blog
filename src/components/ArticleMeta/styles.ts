import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
