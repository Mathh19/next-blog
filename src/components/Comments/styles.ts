import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const NoComments = styled.div`
  ${({ theme }) => css`
    text-align: center;
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.medium};
  `}
`;
