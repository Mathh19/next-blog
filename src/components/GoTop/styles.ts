import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: #082032;
    color: ${theme.colors.secondary};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
    border: 0.2rem solid ${theme.colors.secondary};
    &:hover {
      transition: 500ms ease-in-out all;
      background-color: #12324a;
      opacity: unset;
    }
  `}
`;
