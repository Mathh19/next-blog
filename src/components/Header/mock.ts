import { HeaderProps } from '.';

export default {
  srcImg: 'assets/images/cat-test.png',
  title: 'Titulo blog',
  description:
    'Ea aliqua labore anim minim voluptate est in cillum qui consequat cupidatat adipisicing ex in. Amet velit consequat mollit deserunt consectetur dolor exercitation commodo consequat voluptate eu non commodo ipsum.',
  showDescription: true,
} as HeaderProps;
