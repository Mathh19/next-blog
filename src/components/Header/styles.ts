import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: ${theme.sizes.max};
  padding-bottom: ${theme.spacings.xhuge};
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.darkText};

  ${HeadingStyles} {
    margin: 0 0 0.6rem;
  }
  `}

  img:hover {
    transition: all ease-in-out 300ms;
    filter: brightness(70%);
  }
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
