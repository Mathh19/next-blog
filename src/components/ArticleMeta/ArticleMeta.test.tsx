import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from '.';
import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(
      screen.getByRole('link', { name: 'Matheus Freitas' }),
    ).toHaveAttribute('href', '/author/matheus-freitas');
    expect(screen.getByRole('link', { name: 'React' })).toHaveAttribute(
      'href',
      '/category/react',
    );
    expect(screen.getByRole('link', { name: 'JavaScript' })).toHaveAttribute(
      'href',
      '/category/javascript',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByText('2 de set. de 2022')).toHaveAttribute(
      'datetime',
      props.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should no author and no categories', () => {
    const { container } = renderTheme(
      <ArticleMeta {...props} author={undefined} categories={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
