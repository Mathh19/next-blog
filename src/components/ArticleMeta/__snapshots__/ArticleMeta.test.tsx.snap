// Vitest Snapshot v1

exports[`<ArticleMeta /> > should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #E94560;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <article
    class="c0"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/matheus-freitas"
      >
        Matheus Freitas
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2022-09-02T15:16:09.246+00:00"
      >
        2 de set. de 2022
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react"
          >
            React
          </a>
        </span>
        <span>
          <a
            href="/category/javascript"
          >
            JavaScript
          </a>
        </span>
      </span>
    </p>
  </article>
</div>
`;

exports[`<ArticleMeta /> > should no author and no categories 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #E94560;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <article
    class="c0"
  >
    <p>
      <time
        datetime="2022-09-02T15:16:09.246+00:00"
      >
        2 de set. de 2022
      </time>
    </p>
  </article>
</div>
`;
